src/Serialization.js

```diff
@@ -12,13 +12,17 @@ const setter = prop => obj => value => () => obj["set_" + prop](value);
 exports.hashTransaction = body => () => lib.hash_transaction(body);
 
 exports.newBigNum = maybe => string => {
-  try {
-    return maybe.just(lib.BigNum.from_str(string));
-  } catch (_) {
-    return maybe.nothing;
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
+  if (/^-?[0-9]+$/.test(str)) {
```

Comment:
This check is not enough, it does not account for overflows. Please find the boundary sizes for BigNum and ensure that the numbers are in range by converting their string representations to native JS BigInt and comparing with the boundary,

Reply:
removed the regex, as `BigInt(string)` in the boundary check already throws an error if the string is incorrect